src/vcf/validator.cpp

```diff
@@ -132,6 +132,65 @@ namespace ebi
         }
     }
 
+    bool is_compressed_file(std::istream &input,
+                            const std::string &source)
+    {
+        std::string message = "Input file is a compressed file ";
+
+        if (source != ebi::vcf::STDIN) {
+            boost::filesystem::path source_name(source);
+            std::string file_extension = source_name.extension().string();
+
+            if (file_extension == ".zip") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(zip)";
+                return true;
+            }
+            else if (file_extension == ".tar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar)";
+                return true;
+            }
+            else if (file_extension == ".gz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.gz)";
+                return true;
+            }
+            else if (file_extension == ".xz") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(tar.xz)";
+                return true;
+            }
+            else if (file_extension == ".rar") {
+                BOOST_LOG_TRIVIAL(warning) << message << "(rar)";
+                return true;
+            }
+        }
+
+        unsigned char magic[10];
+        input.read((char*)magic, sizeof(magic));
+
+        const unsigned char zip[4] = { 80, 75, 3, 4 };
+        const unsigned char tar_gz[2] = { 31, 139 };
+        const unsigned char tar_xz[5] = { 253, 55, 122, 88, 90 };
+        const unsigned char tar_z[5] = { 31, 157 };
```

Comment:
this is optional, but I don't like having to put the count of elements. this always ends up slowing modifications a lot. to change this, find alternate ways to specify the array, or use a std::vector.

Reply:
I have made a vector of vectors for it and we can easily use for each loop for checking them all